lib/rules/no-empty-buttons-v9.js

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+"use strict";
+
+const { hasTextContentChild } = require("../util/hasTextContentChild");
+var elementType = require("jsx-ast-utils").elementType;
+var hasProp = require("jsx-ast-utils").hasProp;
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+const allowedComponents = ["Button", "ToggleButton", "SplitButton", "MenuButton", "CompoundButton"];
```

Comment:
The compound button is the exception to the rule, as it's not empty if it has secondary content. Maybe we could add an if statement?